e2e/hpa/01-assert.yaml

```diff
@@ -0,0 +1,98 @@
+---
+apiVersion: quay.redhat.com/v1
+kind: QuayRegistry
+metadata:
+  finalizers:
+  - quay-operator/finalizer
+  name: hpa
+spec:
+  components:
+  - kind: clair
+    managed: false
+  - kind: clairpostgres
+    managed: false
+  - kind: base
+    managed: true
+  - kind: postgres
+    managed: true
+  - kind: redis
+    managed: true
+  - kind: horizontalpodautoscaler
+    managed: false
+  - kind: objectstorage
+    managed: true
+  - kind: route
+    managed: true
+  - kind: mirror
+    managed: true
+  - kind: monitoring
+    managed: true
+  - kind: tls
+    managed: true
+status:
+  conditions:
+  - message: Horizontal pod autoscaler not managed by the operator
+    reason: ComponentNotManaged
+    status: "True"
+    type: ComponentHPAReady
+  - message: Route admitted
+    reason: ComponentReady
+    status: "True"
+    type: ComponentRouteReady
+  - message: ServiceMonitor and PrometheusRules created
+    reason: ComponentReady
+    status: "True"
+    type: ComponentMonitoringReady
+  - message: Deployment hpa-quay-database healthy
+    reason: ComponentReady
+    status: "True"
+    type: ComponentPostgresReady
+  - message: Object bucket claim bound
+    reason: ComponentReady
+    status: "True"
+    type: ComponentObjectStorageReady
+  - message: Clair not managed by the operator
+    reason: ComponentNotManaged
+    status: "True"
+    type: ComponentClairReady
+  - message: ClairPostgres not managed by the operator
+    reason: ComponentNotManaged
+    status: "True"
+    type: ComponentClairPostgresReady
+  - message: Using cluster wildcard certs
+    reason: ComponentReady
+    status: "True"
+    type: ComponentTLSReady
+  - message: Deployment hpa-quay-redis healthy
+    reason: ComponentReady
+    status: "True"
+    type: ComponentRedisReady
+  - message: Base component healthy
+    reason: ComponentReady
+    status: "True"
+    type: ComponentBaseReady
+  - message: Deployment hpa-quay-mirror healthy
+    reason: ComponentReady
+    status: "True"
+    type: ComponentMirrorReady
+  - message: All components reporting as healthy
+    reason: HealthChecksPassing
+    status: "True"
+    type: Available
+  - message: All registry components created
+    reason: ComponentsCreationSuccess
+    status: "True"
+    type: ComponentsCreated
+  - message: All objects created/updated successfully
+    reason: ComponentsCreationSuccess
+    status: "False"
+    type: RolloutBlocked
+---
+apiVersion: autoscaling/v1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: hpa-quay-mirror
```

Comment:
This does not seem right. When the QuayRegistry is created we expect to see three HPAs and their names are:

```
X-clair-app
X-quay-app
X-quay-mirror
```
Where X is equal to QuayRegistry.Spec.Name.

Reply:
I've thought that testing against 1 hpa would be enough, would write all 3 of them then.

> This does not seem right. When the QuayRegistry is created we expect to see three HPAs and their names are:
> 
> ```
> X-clair-app
> X-quay-app
> X-quay-mirror
> ```
> 
> Where X is equal to QuayRegistry.Spec.Name.